src/openzeppelin/token/erc721/erc721.cairo

```diff
@@ -0,0 +1,411 @@
+#[contract]
+mod ERC721 {
+    // OZ modules
+    use openzeppelin::account;
+    use openzeppelin::introspection::erc165;
+    use openzeppelin::token::erc721;
+
+    // Dispatchers
+    use openzeppelin::introspection::erc165::IERC165Dispatcher;
+    use openzeppelin::introspection::erc165::IERC165DispatcherTrait;
+    use super::super::interface::IERC721ReceiverDispatcher;
+    use super::super::interface::IERC721ReceiverDispatcherTrait;
+
+    // Other
+    use starknet::ContractAddress;
+    use starknet::get_caller_address;
+    use zeroable::Zeroable;
+    use option::OptionTrait;
+    use array::SpanTrait;
+    use traits::Into;
+    use openzeppelin::utils::serde::SpanSerde;
+
+    struct Storage {
+        _name: felt252,
+        _symbol: felt252,
+        _owners: LegacyMap<u256, ContractAddress>,
+        _balances: LegacyMap<ContractAddress, u256>,
+        _token_approvals: LegacyMap<u256, ContractAddress>,
+        _operator_approvals: LegacyMap<(ContractAddress, ContractAddress), bool>,
+        _token_uri: LegacyMap<u256, felt252>,
+    }
+
+    #[event]
+    fn Transfer(from: ContractAddress, to: ContractAddress, token_id: u256) {}
+
+    #[event]
+    fn Approval(owner: ContractAddress, approved: ContractAddress, token_id: u256) {}
```

Comment:
as discussed offline, it doesn't seem we need to support camelCase in event keys, since they have no low-level encoding impact, thus not part of the interface. #614 is outside scope of this PR.